ibis/expr/rules.py

```diff
@@ -294,12 +295,35 @@ def _promote_integral_binop(exprs, op):
     return dt.highest_precedence(dtypes)
 
 
+def _infer_decimal_dtype(args, op):
+    # TODO: Add support for the maximum precision and maximum scale
+    # TODO: Currently assumes that there are only two args. Can there be more?
```

Comment:
I suppose in the case of `ibis.coalesce(*only_decimal_args)` there are a variable number of decimal arguments, but the rules for computing the precision and scale of such an expression are going to be different than those here.

We don't have any arithmetic operations that are modeled as variadic, but it's an idea we've been considering in an effort to make pattern matching and simple expression optimizations easier to encode. If we did do that then it'd be in 4.0.

Reply:
Okay, I see. I changed the code s.t. it checks that there are exactly 2 args.

Assuming that the semantics of a variadic add is something like "sum all", we could probably just use the same functions to compute the precision and scale in a pairwise manner, i.e. take the first two, compute their precision/scale, add the result to the list, take the next two... until the list has exactly one element.

Reply:
Yeah, that makes sense to me.